packages/api/portfolios/createPortfolio.ts

```diff
@@ -0,0 +1,86 @@
+import "reflect-metadata";
+import cors from "@middy/http-cors";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import middy from "@middy/core";
+import validator from "@middy/validator";
+import xssSanitizer from "../portfolioDrafts/xssSanitizer";
+import { ApiGatewayEventParsed } from "../utils/eventHandlingTool";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../utils/response";
+import { CORS_CONFIGURATION } from "../utils/corsConfig";
+import { createConnection } from "../utils/database";
+import { DATABASE_ERROR } from "../utils/errors";
+import { errorHandlingMiddleware } from "../utils/errorHandlingMiddleware";
+import { IpCheckerMiddleware } from "../utils/ipLogging";
+import { IPortfolio, IPortfolioCreate } from "../../orm/entity/Portfolio";
+import { PortfolioRepository } from "../repository/PortfolioRepository";
+import { validateRequestShape } from "../utils/shapeValidator";
+import { wrapSchema } from "../utils/schemaWrapper";
+import internalSchema = require("../models/internalSchema.json");
+
+/**
+ * Creates a new Portfolio
+ * A new portfolioId will be generated by the database.
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IPortfolio>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  validateRequestShape<IPortfolio>(event);
+
+  // set up database connection
+  const connection = await createConnection();
+  const insertedPortfolios: IPortfolioCreate[] = [];
+
+  try {
+    const setupResult = validateRequestShape<IPortfolio>(event);
```

Comment:
Doing this twice? see line 32

Reply:
Good catch. Fixed in 2b24834ba61dbf1e311de8432f4c80ed09cf7337